src/server.h

```diff
@@ -57,6 +57,10 @@
 #include <systemd/sd-daemon.h>
 #endif
 
+#ifndef static_assert
+#define static_assert(expr, lit) extern char __static_assert_failure[(expr) ? 1:-1]
```

Comment:
Is this available on older compilers?

Reply:
This is meant for older compilers only (pre c11) because static_assert is not defined there.

Reply:
Sure, my question is how do we know it's defined everywhere? A quick google search didn't show that much. Is this like a standard trick?

Reply:
Would [this link](https://en.cppreference.com/w/c/language/_Static_assert) help? It is part of the c11 spec.

>This keyword is also available as convenience macro [static_assert](https://en.cppreference.com/w/c/error/static_assert), available in the header <assert.h>.

Reply:
Sure, but we also support systems that don't run on C11, do we know this runs on like C99? We've had people in the past complain that there were compilation errors.

Reply:
Right. This change is exactly to make sure static_assert works on all versions of c compilers - notice the n in #if**n**def.

Reply:
Yes, I see that, I'm asking how we know the other function you're aliasing to, "__static_assert_failure" is defined?

Reply:
I see. That is not a function. It is a global variable name that does not have any storage associated with it (because of the "extern" keyword). It is actually a well established way to emulate the c11 static_assert behavior on older compilers. If the expr evaluates false, this macro will be expanded into an array of negative size thus breaking the build. If the expr evaluates true, this macro is essentially a noop since it is declared with "extern".

Reply:
Thanks for the explanation :)